### PR TITLE
feat: divide backend-ai-session-launcher into several pages.

### LINKED ISSUE #915: Apply carousel UI in current session-launcher
**Is your feature request related to a problem? Please describe.**
기능 요청이 특정 문제에 연관된 것이라면 여기에 그 문제를 설명해주세요.
There have been a lot of updates and bugfixes since Backend.AI released, thereafter lots of settings are added in the "session launcher", which brought a strong necessity to split each configuration to provide the user to double-check their settings before the launching.
For example, if there's any chance of having entities more than 15 in Vfolder selection, the height of the session launcher dialog goes so stretched that It ruins UX.
To solve those problems, and enhance the user experiences, each of the sections should be split into at most 4 pages.

**Describe the solution you'd like**
어떤 기능이 있으면 좋겠는지 자세히 설명해주세요.
For now, there is a single page that contains whole configurations. Although those configurations are related to each other, They can be clustered in three parts, according to the priority of each feature.
- First page
   - Environment
   - Version
   - Session name
   - Environment Variable settings

- Second page
   - Folder to mount

- Third page
   - Resource Group
   - Resource Allocation (including Cluster mode)
   - Session owner* (SuperAdmin only)

- Fourth page (The last page)
   - Environment and versions
   - Session name (optional)
   - Environment Variable settings (optional)
   - Mounted Folders (including Automount folders)
   - Total Resource Allocation
   - Session owner (optional)
   - Launch button

**Describe alternatives you've considered**
혹시 다른 대안들을 생각해본 적이 있다면 함께 적어주세요.

**Additional context**
기능 요청에 대해 보다 잘 이해할 수 있는 다른 맥락을 기술해주세요.

**Screenshots**
too_long_session_launcher

### PR DESCRIPTION
**Pull request by Sujin-Kim1**
- Relocate the dialog content of backend-ai-session-launcher.ts.
- And add previous, next buttons to move progress.

this closes #915

### CODE REVIEW

File: src/components/backend-ai-session-launcher-new.ts [hunk 1]
```diff
@@ -0,0 +1,3082 @@
...
+        .resource-allocated-box {
+          background-color: var(--paper-grey-300);
+          border-radius: 5px;
+          margin: 5px;
+          z-index:10;
+        }
+
+        #new-session-dialog {
+          --component-width: 400px;
+          --component-max-height: 640px;
```
Comment (lizable) [inline src/components/backend-ai-session-launcher-new.ts]: We need to specify the maximum size of the dialog to provide a better user experience in the mobile environment. Therefore, I added the maximum height of the session-launcher dialog.

File: src/components/backend-ai-session-launcher-new.ts [hunk 2]
```diff
@@ -0,0 +1,3082 @@
...
+        }
+        mwc-select#owner-scaling-group {
+          margin-right: 0;
+          padding-right: 0;
+          width: 50%;
+          --mdc-select-min-width: 190px;
+        }
+
+        mwc-select > mwc-list-item {
+          width: 370px; // default width
```
Comment (lizable) [inline src/components/backend-ai-session-launcher-new.ts]: I added the width of `mwc-list-item`, which is the children element of `mwc-select` and `mwc-list`. It follows the width of those elements.
please keep in mind that when you added `fixedMenuPosition` attribute in `mwc-select`, this style setting becomes necessary.
(ref: [`mwc-select` documentation](https://github.com/material-components/material-components-web-components/tree/master/packages/select))

File: src/components/backend-ai-session-launcher-new.ts [hunk 3]
```diff
@@ -0,0 +1,3082 @@
...
+      <wl-button raised class="primary-action" id="launch-session" ?disabled="${!this.enableLaunchButton}"
+                 @click="${() => this._launchSessionDialog()}">
+        <wl-icon>power_settings_new</wl-icon>
+        <span>${_t('session.launcher.Start')}</span>
+      </wl-button>
+      <backend-ai-dialog id="new-session-dialog" narrowLayout fixed backdrop persistent>
+        <span slot="title">${this.newSessionDialogTitle ? this.newSessionDialogTitle : _t('session.launcher.StartNewSession')}</span>
+        <form slot="content" id="launch-session-form" class="centered" style="position:relative;">
+          <div id="progress-01" class="progress center layout fade active">
+            <mwc-select id="environment" icon="code" label="${_t('session.launcher.Environments')}" required fixedMenuPosition
```
Comment (lizable) [inline src/components/backend-ai-session-launcher-new.ts]: When you add `fixedMenuPosition` attribute in the `mwc-select` element, item elements will be exposed in front of the dialog if `mwc-select` is inside of `mwc-dialog`.

File: src/components/backend-ai-session-launcher-new.ts [hunk 4]
```diff
@@ -0,0 +1,3082 @@
...
+    this._showKernelDescription(e, item);
+  }}">
+                        </mwc-icon-button>
+                      </div>
+                    </div>
+                  </mwc-list-item>
+                `}
+              `)}
+            </mwc-select>
+            <mwc-select id="version" icon="architecture" label="${_t('session.launcher.Version')}" required fixedMenuPosition>
```
Comment (lizable) [inline src/components/backend-ai-session-launcher-new.ts]: Same explanation from above.

File: src/components/backend-ai-session-launcher-new.ts [hunk 5]
```diff
@@ -0,0 +1,3082 @@
...
+                    id="unselect-all-vfolders"
+                    style="width:auto;margin-right:10px;"
+                    @click=${() => this._unselectAllSelectedFolder()}></mwc-button>
+              </div>
+            ` : html``}
+          </div>
+          <div id="progress-03" class="progress center layout fade">
+            <div class="horizontal center layout">
+              <mwc-select id="scaling-groups" label="${_t('session.launcher.ResourceGroup')}" 
+                          icon="storage" required fixedMenuPosition
```
Comment (lizable) [inline src/components/backend-ai-session-launcher-new.ts]: Same explanation from above.

File: src/components/backend-ai-session-launcher-new.ts [hunk 6]
```diff
@@ -0,0 +1,3082 @@
...
+                                id="${item.name}" graphic="icon"
+                                value="${item.name}">
+                    ${item.name}
+                  </mwc-list-item>
+                `)}
+              </mwc-select>
+            </div>
+            <div class="vertical center layout" style="position:relative;">
+              <mwc-select id="resource-templates" label="${_t('session.launcher.ResourceAllocation')}"
+                          icon="dashboard_customize" required fixedMenuPosition>
```
Comment (lizable) [inline src/components/backend-ai-session-launcher-new.ts]: Same explanation from above.

File: src/components/backend-ai-session-launcher-new.ts [hunk 7]
```diff
@@ -0,0 +1,3082 @@
...
+                  <span class="caption">#</span>
+                  <mwc-icon-button icon="info" class="fg red info" @click="${(e) => {
+    this._showResourceDescription(e, 'session');
+  }}"></mwc-icon-button>
+                </div>
+              </div>
+            </wl-expansion>
+            ${this.cluster_support ? html`
+              <mwc-select id="cluster-mode" label="${_t('session.launcher.ClusterMode')}" required
+                          icon="account_tree" fixedMenuPosition
```
Comment (lizable) [inline src/components/backend-ai-session-launcher-new.ts]: Same explanation from above.

File: src/components/backend-ai-session-launcher-new.ts [hunk 8]
```diff
@@ -0,0 +1,3082 @@
...
+              <div class="vertical layout">
+                <div class="horizontal center layout">
+                  <mwc-textfield id="owner-email" type="email" class="flex" value=""
+                                pattern="^.+@.+\..+$" icon="mail"
+                                label="${_t('session.launcher.OwnerEmail')}" size="40"></mwc-textfield>
+                  <mwc-icon-button icon="refresh" class="blue"
+                                  @click="${() => this._fetchSessionOwnerGroups()}">
+                  </mwc-icon-button>
+                </div>
+                <mwc-select id="owner-accesskey" label="${_t('session.launcher.OwnerAccessKey')}" icon="vpn_key" fixedMenuPosition>
```
Comment (lizable) [inline src/components/backend-ai-session-launcher-new.ts]: Same explanation from above.

File: src/components/backend-ai-session-launcher-new.ts [hunk 9]
```diff
@@ -0,0 +1,3082 @@
...
+                  ${this.ownerKeypairs.map((item) => html`
+                    <mwc-list-item class="owner-group-dropdown"
+                                  id="${item.access_key}"
+                                  value="${item.access_key}">
+                      ${item.access_key}
+                    </mwc-list-item>
+                  `)}
+                </mwc-select>
+                <div class="horizontal center layout">
+                  <mwc-select id="owner-group" label="${_t('session.launcher.OwnerGroup')}" icon="group_work" fixedMenuPosition>
```
Comment (lizable) [inline src/components/backend-ai-session-launcher-new.ts]: Same explanation from above.

File: src/components/backend-ai-session-launcher-new.ts [hunk 10]
```diff
@@ -0,0 +1,3082 @@
...
+                    ${this.ownerGroups.map((item) => html`
+                      <mwc-list-item class="owner-group-dropdown"
+                                    id="${item.name}"
+                                    value="${item.name}">
+                        ${item.name}
+                      </mwc-list-item>
+                    `)}
+                  </mwc-select>
+                  <mwc-select id="owner-scaling-group" label="${_t('session.launcher.OwnerResourceGroup')}"
+                              icon="storage" fixedMenuPosition>
```
Comment (lizable) [inline src/components/backend-ai-session-launcher-new.ts]: Same explanation from above.

**Review by adrysn — CHANGES_REQUESTED** [review]
I just used the PR and unable to launch a session. Please refer to the following image.
image

The vertical align mismatch for the previous/next button and the arrows. And, how about using a bigger font size for the prev/next button?
image

Personally, I think it would be good to have another button that states "LAUNCH NOW" in the middle of the prev and next buttons. When a user clicks this button, the session is launched with the default settings without visiting the next dialog contents. For example, when you launch an instance in AWS, you can see a button like "검토 및 시작"
image

**Comment by inureyes** [comment]
So many items with same id. It SHOULD BE UNIQUE. 

e.g. span id="launch-button-msg"

**Comment by lizable** [comment]
> So many items with same id. It SHOULD BE UNIQUE.
> 
> e.g. span id="launch-button-msg"

I guess `launch-btn-msg` is only meaningful in `new-session-dialog`.
The rest of id declarations seems redundant to me. I'll remove those id declarations except in `new-session-dialog` right away.

**Review by inureyes — CHANGES_REQUESTED** [review]
* What if the set environment variable appears below?
* How about adding indicator below the Review and Start buttons to let users know what stage they are on.
* In the DOM, the ID must be single.

**Review by inureyes — APPROVED** [review]
LGTM

**Pull request merged**